### DISCUSSION TITLE
improve rendering shader linked by default in the build of game

### DISCUSSION THREAD

**Discussion by catafest** [godotengine/godot-proposals#7787] [3D]
### Describe the project you are working on

I try to create a good ocean in Godot.
In Unity, 3D has a default implementation.
I suppose the reason is the implementation in programming and build result and final game render result.

### Describe the problem or limitation you are having in your project

I'm not a c++ programmer but I tested the CG programming language and I think the Godot team can used the real programming source code and not GDShaders.C++programmer

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The good idea is to use tools and default implementations for users like any engine: shaders for 2D and 3D for basic elements fog, water, ocean halo, grass, fire, and more. This can easily add to a project if in basic shader programming like CG or another shader programming. Now the GDshader implementation is strange and without good documentation.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

use a Godot node for the ocean in the project with a good cg shader or at least a good gdshader for Godot version 4.1.x 

### If this enhancement will not be used often, can it be worked around with a few lines of script?

the idea to add a default shader in the build can help the render for 2D and 3D shaders and can improve it.

### Is there a reason why this should be core and not an add-on in the asset library?

yes, because this can be linked with default build and the render of the game the

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/6900.

Moving to discussion, as there is no concrete technical implementation provided.

High-level shaders are generally better provided by add-ons, as we intend to keep the core feature set highly general-purpose and not specialized to a certain type of game or art style. It also allows making people only pay for the features they actually use (in terms of binary size).
